test/unit-tests/broker-launch.test.js

```diff
@@ -1,26 +1,66 @@
+const sinon = require('sinon');
+const cp = require('child_process');
+const { EventEmitter } = require('events');
 const expect = require('chai').expect;
-const io = require('socket.io-client');
 
-describe('Broker connection handling', () => {
-    let client, dashboardServer;
+const launchBroker = require('../../dashboard/index');
 
-    before((done) => {
-        dashboardServer = require('../../dashboard/server/index');
-        client = io('http://localhost:5001/');
-        client.on('connect', done);
+// mock event emitter to simulate ChildProcess object
+const mockChildProcess = new EventEmitter();
+
+// sinon spies for behaviour verification
+let unrefSpy = sinon.spy();
+let disconnectSpy = sinon.spy();
+
+describe('Broker daemonized launching', () => {
+    before(() => {
+        // stub cp.spawn, to avoid actual daemon launch
+        sinon.stub(cp, 'spawn').callsFake(() => {
+            mockChildProcess['unref'] = unrefSpy;
+            mockChildProcess['disconnect'] = disconnectSpy;
+
+            return mockChildProcess;
+        });
+
+        // stub process.exit to avoid exiting
+        sinon.stub(process, 'exit');
+
+        // stub console.log for daemon status output
+        sinon.stub(console, 'log');
+
+        launchBroker(5001);
     });
 
     after(() => {
-        // close client and server to exit the CLI
-        client.close();
-        dashboardServer.io.close();
-        dashboardServer.server.close();
+        cp.spawn.restore();
+        process.exit.restore();
+        console.log.restore();
     });
 
-    it('should emit events', (done) => {
-        client.emit('test-conn', (arg) => {
-            expect(arg).to.equal('hello world');
-            done();
+    it('should provide correct spawn arguments', () => {
+        expect(cp.spawn.args[0][0]).to.equal(process.execPath);
+        expect(cp.spawn.args[0][1]).to.eql(['./dashboard/server/index.js']);
+        expect(cp.spawn.args[0][2]).to.eql({
+            detached: true,
+            stdio: ['ignore', 'ignore', 'ignore', 'ipc'],
         });
     });
+
+    it('should launch broker correctly', () => {
+        mockChildProcess.emit('message', (cb) => cb());
+
+        expect(unrefSpy.callCount).to.equal(1);
+        expect(unrefSpy.calledWith()).to.be.true;
+
+        expect(disconnectSpy.callCount).to.equal(1);
+        expect(unrefSpy.calledWith()).to.be.true;
```

Comment:
Maybe a better assertion could be:
```suggestion
        expect(unrefSpy.firstCall.args).to.have.lengthOf(0);
```

Reply:
Done 🚀